packages/connect-form/src/components/connect-form-actions.tsx

```diff
@@ -11,37 +11,76 @@ import {
   ConnectionFormError,
   ConnectionFormWarning,
 } from '../utils/validation';
+import type { ConnectionInfo } from 'mongodb-data-service';
 
 const formActionStyles = css({
   borderTop: `1px solid ${uiColors.gray.light2}`,
   paddingLeft: spacing[4],
   paddingRight: spacing[4],
 });
 
+const formActionButtonStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  '& div:first-child': {
+    marginRight: spacing[2],
+  },
+});
+
 const formButtonsStyles = css({
   paddingTop: spacing[3],
   paddingBottom: spacing[3],
   textAlign: 'right',
 });
 
 function ConnectFormActions({
+  initialConnectionInfo,
   errors,
   warnings,
   onConnectClicked,
+  onSaveClicked,
+  saveDisabled,
 }: {
-  onConnectClicked: () => void;
+  initialConnectionInfo: ConnectionInfo;
   errors: ConnectionFormError[];
   warnings: ConnectionFormWarning[];
+  onConnectClicked: () => void;
+  onSaveClicked: () => void;
+  saveDisabled: boolean;
 }): React.ReactElement {
   return (
     <div className={formActionStyles}>
       {warnings.length ? <WarningSummary warnings={warnings} /> : ''}
       {errors.length ? <ErrorSummary errors={errors} /> : ''}
-
-      <div className={formButtonsStyles}>
-        <Button variant={ButtonVariant.Primary} onClick={onConnectClicked}>
-          Connect
-        </Button>
+      <div className={formActionButtonStyles}>
+        {!initialConnectionInfo.favorite && (
+          <div className={formButtonsStyles}>
+            <Button variant={ButtonVariant.Primary} onClick={onConnectClicked}>
+              Connect
+            </Button>
+          </div>
+        )}
+        {!!initialConnectionInfo.favorite && (
+          <Fragment>
+            <div className={formButtonsStyles}>
+              <Button
+                variant={ButtonVariant.Default}
+                disabled={saveDisabled}
+                onClick={onSaveClicked}
+              >
+                Save
+              </Button>
+            </div>
+            <div className={formButtonsStyles}>
+              <Button
```

Comment:
other nit, is not necessary to copy paste the connect button here, you can have the whole thing like this:

```
{initialConnectionInfo.favorite ?? <button>Save</button>}
<button>Connect</button>
```

Reply:
I removed the `connectionInfo` parameter from this component, because it was used only for that check. Let me know what you suggest to do